Sources/Core/SoftwareBreakpointManager.cpp

```diff
@@ -124,4 +124,30 @@ bool SoftwareBreakpointManager::fillStopInfo(Target::Thread *thread,
   stopInfo.reason = StopInfo::kReasonBreakpoint;
   return true;
 }
+
+void SoftwareBreakpointManager::insertStashedInsns(Address const &start,
+                                                   size_t length,
+                                                   ByteVector &data) {
+  Address end = start + length;
+  for (auto const &insn : _insns) {
+    Address insn_start = insn.first;
+    Address insn_end = insn.first + insn.second.size();
+
+    // no overlap between region and instruction
+    if (end < insn_start || start >= insn_end) {
+      continue;
+    }
+
+    // overlap begins at beginning of instruction
+    if (start <= insn_start) {
+      uint64_t offset = insn_start - start;
+      uint64_t size = std::min((uint64_t)insn.second.size(), end - insn_start);
+      memcpy(&data[offset], &insn.second[0], size);
+    } else {
+      uint64_t offset = start - insn_start;
+      uint64_t size = insn_end - start;
+      memcpy(&data[0], &insn.second[offset], size);
+    }
+  }
+}
```

Comment:
Something I just thought about: Is it possible for and instruction size to be greater than `end - start`? By that, I mean is this possible: `insn_start <= start <= end <= insn_end`?

My gut says it's possible but unlikely. If so, the `size` in the else path will be larger than you're allowed to write and you might segfault. It should be something like `std::min(insn_end - start, end - start)`, I think.